### PR TITLE
Added runRule, isEnabled and setEnabled

### PR DESCRIPTION
**Pull request by rkoshak**
I also changed 'ruleExists` as the RuleManager seems to be more reliable in UI rules than `ruleRegistry`. Sometimes `ruleRegistry` would say a rule doesn't exist when it does buy `RuleManager` always returns a rule's status.

Errors are thrown if the rule doesn't exist in all of these. An error is also thrown if `runRule` is called and the rule is UNINITIALIZED which can be because it's been disabled or is broken somehow).

Signed-off-by: Richard Koshak <rlkoshak@gmail.com>

### CODE REVIEW

File: rules/rules.js [hunk 1]
```diff
@@ -108,6 +108,55 @@ const removeRule = function (uid) {
...
+ * 
+ * @memberOf rules
+ * @param {String} uid the UID of the rule to run
+ * @param {Map<Object>} args optional dict of data to pass to the called rule
+ * @param {Boolean} cond when true, the called rull will only run if it's conditions are met
+ */
+const runRule = function (uid, args = {}, cond = true) {
+    const status = RuleManager.getStatus(uid);
+    if(!status) {
+        throw 'There is no rule with UID ' + uid;
```
Comment (florian-h05) [inline rules/rules.js:123]: May we use `throw Error()` instead of `throw ()` for consistency across openhab-js?

File: rules/rules.js [hunk 2]
```diff
@@ -108,6 +108,55 @@ const removeRule = function (uid) {
     }
 };
 
+/**
+ * Runs the rule with the given UID. Throws errors when the rule doesn't exist
+ * or is unable to run (e.g. it's disabled);
```
Comment (florian-h05) [inline rules/rules.js:113]: I think the the semicolon is a typo and should be a point.

File: rules/rules.js [hunk 3]
```diff
@@ -108,6 +108,55 @@ const removeRule = function (uid) {
...
+ * 
+ * @memberOf rules
+ * @param {String} uid the UID of the rule to run
+ * @param {Map<Object>} args optional dict of data to pass to the called rule
+ * @param {Boolean} cond when true, the called rull will only run if it's conditions are met
```
Comment (florian-h05) [inline rules/rules.js:118]: Typo: should be rule and not rull

Comment (florian-h05) [reply rules/rules.js:118]: `args` and `cond` are optional and have defaults.
You can adjust the JSDoc to show that:
```js
/** 
 * …
 * @param {Map<Object>} [args={}] optional dict …
 * @param {Boolean} [cond=true] when true, …
 */
```

**Review by florian-h05 — CHANGES_REQUESTED** [review]
@rkoshak 
Overall looks good, just a few points.
Please have a look at my comments.

File: rules/rules.js (hunk 2 above)
Comment (rkoshak) [reply rules/rules.js:113]: Note that 109 is not among my changes. That was there to begin with.

I do notice it's not consistent in how it's used in this file so I can change them all to be consistent.

I don't think `.` is right though. It really is the end of a statement, the assignment of the function to `removeRule`.

File: rules/rules.js (hunk 3 above)
Comment (rkoshak) [reply rules/rules.js:118]: I'll fix it.

File: rules/rules.js (hunk 2 above)
Comment (florian-h05) [reply rules/rules.js:113]: I think we are not talking about the same ;-)

> I don't think . is right though. It really is the end of a statement, the assignment of the function to removeRule.

That’s correct.

I mean the semicolon at the end of the comment in line 113.

File: rules/rules.js (hunk 1 above)
Comment (rkoshak) [reply rules/rules.js]: Done

File: rules/rules.js (hunk 3 above)
Comment (rkoshak) [reply rules/rules.js]: Done

**Comment by rkoshak** [comment]
The three changes requested have been made. Pending any other comments this PR is good to go.

File: rules/rules.js (hunk 3 above)
Comment (florian-h05) [reply rules/rules.js]: Did you fix that? I can‘t see changes.

**Comment by florian-h05** [comment]
Sign-Offs are missing, I don‘t know how important they are.

Maybe you can add the missing signoffs to the commits or revert the commits till the first one, commit again in one commit with all new changes and then force push (that‘s my way of going when I want to fix something in a previous commit).

**Comment by rkoshak** [comment]
I added one to the most recent commit but researching how to add it to the one before that.

**Comment by Confectrician** [comment]
There is an explanation on the checks page for what is wrong and how to solve it:
https://github.com/openhab/openhab-js/pull/81/checks

**Comment by rkoshak** [comment]
I tried the `git rebase --signoff HEAD~4` followed by the `git push --force-with-lease origin runRule` but it doesn't appear to have kicked off the DCO checks again. Or it didn't work, though I did get "Successfully rebased and updated" on my end.

**Comment by rkoshak** [comment]
Well none of that worked. Guess I'll try reverting and starting over.

**Comment by rkoshak** [comment]
This is all so now hopelessly broken I'll just start over with a new PR. It let me revert a couple of commits but it won't let me do it for others. Now I'm getting errors on pushes and pulls. New PR is #82. Closing this one. All the above comments are addressed in #82.

Sorry for the confusion. Don't commit before your first cup of coffee is the lesson I've learned.

**Pull request closed without merging**